File: designpatterns/java/src/decorator/starbuzz/Beverage.java
package decorator.starbuzz;

public abstract class Beverage {

    String description = "제목 없음";

    public enum Size { TALL, GRANDE, VENTI };

    Size size = Size.TALL;

    public String getDescription() {
        return description;
    }

    public Size getSize() {
        return this.size;
    }

    public void setSize(Size size) {
        this.size = size;
    }

    public abstract double cost();
}

File: designpatterns/kotlin/designpatterns/src/main/kotlin/decorator/solve/Beverage.kt
package decorator.solve

abstract class Beverage(
    open val description: String
) {
    abstract fun cost(): Int
}

class HouseBlend(): Beverage("HouseBlend") {
    override fun cost(): Int = 1_000
}

class DarkRoast(): Beverage("DarkRoast") {
    override fun cost(): Int = 2_000
}